dustmite.d

```diff
@@ -1312,20 +1312,20 @@ void dumpSet(string fn)
 				"[",
 				e.noRemove ? "!" : "",
 				" ",
-				e.isFile ? e.filename ? printableFN(e.filename) ~ " " : null : e.head ? printable(e.head) ~ " " : null,
-				e.tail ? printable(e.tail) ~ " " : null,
-				e.comment ? "/* " ~ e.comment ~ " */ " : null,
+				e.isFile ? e.filename !is null ? printableFN(e.filename) ~ " " : null : e.head !is null ? printable(e.head) ~ " " : null,
+				e.tail !is null ? printable(e.tail) ~ " " : null,
+				e.comment !is null ? "/* " ~ e.comment ~ " */ " : null,
```

Comment:
No, this code is correct. The debug output must show when something is null or empty because it affects the behavior in other places.  If it was changed here, it would also need to be changed everywhere else, otherwise the debug output would become misleading.